dids/diddht/internal/dns/did.go

```diff
@@ -141,12 +141,20 @@ func MarshalVerificationMethod(vm *didcore.VerificationMethod) (string, error) {
 		return "", errors.New("unsupported algorithm")
 	}
 
+	// TODO: clean this up. smol monkey patch that accommodates
+	// the possibility that vm.ID may or may not have a '#'.
+	// Technically it always should but i can't confirm that given that
+	// this fix is blocking other work. did:dht will be refactored in
+	// a subsequent PR  soon (Moe: 2024-04-16)
+	splitID := strings.Split(vm.ID, "#")
+	fragment := splitID[len(splitID)-1]
```

Comment:
Here to plug how I did this over in web5-rs via `KeyIdFragment` https://github.com/TBD54566975/web5-rs/blob/9dcdadafe027fea633e0b9ae65985c2931827733/crates/dids/src/document.rs#L45C1-L60C2